### PR TITLE
Features, bugfixes and adjustments on datepicker

### PR DESCRIPTION
**Pull request by guilhermejcgois**

### CODE REVIEW

File: libs/shell/controls/src/lib/ui/datepicker/datepicker.component.html [hunk 1]
```diff
@@ -4,6 +4,28 @@
   [withoutOutput]="true">
   <mat-list role="list" class="inputs">
     <form [formGroup]="form">
+      <mat-list-item role="listitem" class="checked-config-container">
+        <mat-checkbox
+          class="full-width"
+          color="primary"
+          formControlName="activeOnSelect"
+          >Toggle active on date select</mat-checkbox
+        >
```
Comment (oliveira-gust) [inline libs/shell/controls/src/lib/ui/datepicker/datepicker.component.html]: Closing > looks so lonely all by itself here...

File: libs/ui-toolkit/datepicker/src/lib/datepicker-toggle.component.scss [hunk 2]
```diff
@@ -25,5 +25,13 @@
...
+  background-image: linear-gradient(45deg, $text-gray, $textTernary);
+  
+  &:hover {
+    background-image: linear-gradient(45deg, $GorilaDefault, $GorilaSecondary);
+  }
 }
+
+:host.mat-datepicker-toggle-active .mat-icon-button {
+  background: $GorilaDefault;
+}
```
Comment (oliveira-gust) [inline libs/ui-toolkit/datepicker/src/lib/datepicker-toggle.component.scss]: Missing newline at end of file

File: libs/ui-toolkit/datepicker/src/lib/datepicker-toggle.component.ts [hunk 3]
```diff
@@ -50,4 +57,34 @@ export class DatepickerToggleComponent<D> extends MatDatepickerToggle<D> {
   @HostListener('focus') public setFocus() {
     this._button.focus();
   }
+
+  public ngOnInit() {
```
Comment (oliveira-gust) [inline libs/ui-toolkit/datepicker/src/lib/datepicker-toggle.component.ts]: Does MatDatepickerToggle have a ngOnInit method? Is it necessary to call it here as well?

File: libs/ui-toolkit/datepicker/src/lib/datepicker.component.ts [hunk 4]
```diff
@@ -21,6 +21,21 @@ export const SELECTOR = 'gor-datepicker';
   encapsulation: ViewEncapsulation.None
 })
 export class DatepickerComponent<D> extends MatDatepicker<D> {
+  /**
+   * The text to main button of datepicker
+   */
+  @Input() public applyText: string;
+
+  /**
+   * The maximun range date to be selected.
```
Comment (oliveira-gust) [inline libs/ui-toolkit/datepicker/src/lib/datepicker.component.ts]: Typo in comment

File: libs/ui-toolkit/datepicker/src/lib/datepicker.component.ts [hunk 5]
```diff
@@ -43,8 +58,20 @@ export class DatepickerComponent<D> extends MatDatepicker<D> {
     this[p] = v;
   }
 
+  public cloneStartAt() {
```
Comment (oliveira-gust) [inline libs/ui-toolkit/datepicker/src/lib/datepicker.component.ts]: Missing doc for public method

File: libs/ui-toolkit/datepicker/src/lib/datepicker-toggle.component.ts (hunk 3 above)
Comment (guilhermejcgois) [reply libs/ui-toolkit/datepicker/src/lib/datepicker-toggle.component.ts]: Don't have it

**Pull request merged**